src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -135,26 +147,34 @@ const DropdownMenu = <T,>({
         togglePopup();
     };
 
+    const {onKeyDown: handleSwitcherKeyDown} = useActionHandlers(handleSwitcherClick);
+
     return (
         <DropdownMenuContext.Provider value={contextValue}>
-            {/*as this div has Button component as child, clicking on it one fires onClick of this div on bubbling*/}
+            {/* FIXME remove switcher prop and this wrapper */}
             {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
             <div
                 ref={anchorRef}
```

Comment:
The ref is passed twice, yes it will return the second element, but is it better to add a check like this `Boolean(render Switcher) ? undefined : controlRef`